ChainGenerator/Components/Widgets/WidgetGenerator.razor

```diff
@@ -1,6 +1,25 @@
 ﻿@using System.Text
 @inject IServiceProvider ServiceProvider
 
+<!-- The badge icon which indicates an ongoing refresh -->
+@if (refreshInProgress)
+{
+    <MudBadge Origin="Origin.TopLeft"
+              Color="Color.Warning"
+              Overlap="true"
+              Bordered="false"
+              Icon="@badgeIcon">
+    </MudBadge>
+}
+else //Uses a transparent badge to keep the card formatting consistent 
+{
+    <MudBadge Origin="Origin.TopLeft"
+              Color="Color.Transparent"
+              Overlap="true"
+              Bordered="false">
+    </MudBadge>
+}
```

Comment:
This is a great start and heading in the right direction but looking at the documentation I believe this component should **wrap** the card component. That is probably why you were facing issues with the formatting being inconsistent

```
    <MudBadge Origin="Origin.TopLeft"
              Visible="refreshInProgress"   <--- replacing color parameter
              Overlap="true"    <--- not sure if this would still be needed
              Bordered="false"
              Icon="@badgeIcon">

        <!-- The main card that contains the widget generator -->
        <MudCard Outlined="true">
               .....
        </MudCard>

    </MudBadge>
```

There also looks like there is a `Visible` property that we can take advantage of on the badge component
- https://mudblazor.com/api/badge#pages